src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1178,6 +1220,60 @@ impl LinkCollector<'_, '_> {
         }
     }
 
+    fn resolve_with_disambiguator_cached(
+        &mut self,
+        key: CacheKey,
+        item: &Item,
+        dox: &str,
+        current_item: &Option<String>,
+        ori_link: &str,
+        link_range: Option<Range<usize>>,
+        link_ty: LinkType,
+    ) -> Option<(Res, Option<String>)> {
+        // Try to look up both the result and the corresponding side channel value
+        if let Some(ref cached) = self.visited_links.get(&key) {
```

Comment:
`ref` isn't necessary here since it's already a borrow.
```suggestion
        if let Some(cached) = self.visited_links.get(&key) {
```